enterprise/reporting/reporting.go

```diff
@@ -95,28 +99,34 @@ func (handle *HandleT) setup(beConfigHandle backendconfig.BackendConfig) {
 
 	for beconfig := range ch {
 		config := beconfig.Data.(map[string]backendconfig.ConfigT)
-		handle.workspaceIDForSourceIDMapLock.Lock()
 		newWorkspaceIDForSourceIDMap := make(map[string]string)
+		newPIIReportingSettings := make(map[string]bool)
 		var newWorkspaceID string
 
 		for workspaceID, wConfig := range config {
 			newWorkspaceID = workspaceID
 			for _, source := range wConfig.Sources {
 				newWorkspaceIDForSourceIDMap[source.ID] = workspaceID
 			}
+			newPIIReportingSettings[workspaceID] = wConfig.Settings.DataRetention.DisableReportingPII
 		}
 		if len(config) > 1 {
 			newWorkspaceID = ""
 		}
 		handle.workspaceID = newWorkspaceID
 		handle.workspaceIDForSourceIDMap = newWorkspaceIDForSourceIDMap
-		handle.workspaceIDForSourceIDMapLock.Unlock()
+		handle.piiReportingSettings = newPIIReportingSettings
+		handle.onceInit.Do(func() {
+			close(handle.init)
+		})
 	}
+
+	handle.onceInit.Do(func() {
+		close(handle.init)
+	})
 }
 
 func (handle *HandleT) getWorkspaceID(sourceID string) string {
-	handle.workspaceIDForSourceIDMapLock.RLock()
-	defer handle.workspaceIDForSourceIDMapLock.RUnlock()
```

Comment:
let's make sure we have at least received the first configuration
`<-handle.init`

Reply:
Done